crates/biome_css_analyze/src/analyzers/nursery/color_no_invalid_hex.rs

```diff
@@ -0,0 +1,72 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_css_syntax::CssDeclarationOrRuleBlock;
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Succinct description of the rule.
+    ///
+    /// Put context and details about the rule.
+    /// As a starting point, you can take the description of the corresponding _ESLint_ rule (if any).
+    ///
+    /// Try to stay consistent with the descriptions of implemented rules.
+    ///
+    /// Add a link to the corresponding stylelint rule (if any):
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// p {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// p {
+    ///   color: red;
+    /// }
+    /// ```
+    ///
+    pub ColorNoInvalidHex {
```

Comment:
At the beginning the convention made sense because we had few query types, but now I don't think it makes sense anymore. I think we should just kill the convention and put all the rules in the same folder. What do you think?

Reply:
I totally agree that idea since that convention confuses some contributors sometimes. I often see that the rule uses `Ast` is putted in the `/semantic_analyzer`...

Reply:
@ematipico What about lint rule name convention? Should we adhere to the rule names in Stylelint?

Reply:
Rule names should adhere to **our** convention :)